impl11/ddraw/SteamVR.cpp

```diff
@@ -298,96 +299,81 @@ void GetSteamVRPositionalData(float* yaw, float* pitch, float* roll, float* x, f
 	{
 		// Pose array predicted for current frame N, to use by ddraw to render current frame in GPU (minimize latency)
 		vr::TrackedDevicePose_t trackedDevicePoseArray[vr::k_unMaxTrackedDeviceCount];
-		// Pose array predicted for next frame N+1, to use by XWA for CPU calculations next time CockpitLook is called.
-		vr::TrackedDevicePose_t trackedDevicePoseArrayNext[vr::k_unMaxTrackedDeviceCount];
 		vr::TrackedDevicePose_t trackedDevicePose; // HMD pose to use for the current frame render
-		vr::HmdMatrix34_t m34_correctionMatrix;
+		vr::HmdMatrix34_t m34_fullMatrix;
 		vr::HmdQuaternionf_t q;
 		vr::ETrackedDeviceClass trackedDeviceClass = vr::VRSystem()->GetTrackedDeviceClass(unDevice);
 
-		vr::VRCompositor()->WaitGetPoses(trackedDevicePoseArray, vr::k_unMaxTrackedDeviceCount, trackedDevicePoseArrayNext, vr::k_unMaxTrackedDeviceCount);
-
-		// When the mouse hook is active, we the pose through g_pSharedData; but in 2D mode, or when we
-		// are in the hangar, we still need to query SteamVR directly. So we have to check g_bRendering3D
-		// and g_playerInHangar to enable head tracking in 2D mode or when in the hangar.
-		//if (g_bRendering3D && !*g_playerInHangar && (g_pSharedData != NULL && g_pSharedData->bDataReady)) {
-		if (g_bRendering3D && !(*g_playerInHangar)) {
-			// Get the last tracking pose obtained by CockpitLook. This pose was just used to render the
-			// current frame in xwingaliance.exe.
-			// We can also make a global vr::TrackedDevicePose_t* and initialize it to g_pSharedData->pDataPtr once
-			// bDataReady is true, but I don't think there's a huge advantage to doing that.
-			//trackedDevicePose = *(vr::TrackedDevicePose_t*) g_pSharedData->pDataPtr;
-			//log_debug("[DBG] Using trackedDevicePose from CockpitLook\n");
-			//vr::VRCompositor()->GetLastPoses(trackedDevicePoseArray, vr::k_unMaxTrackedDeviceCount, NULL, 0);
-			trackedDevicePose = trackedDevicePoseArray[vr::k_unTrackedDeviceIndex_Hmd];
+		if (g_bRendering3D && !(*g_playerInHangar) && !g_bCorrectedHeadTracking) {
+			// For legacy/stable tracking, WaitGetPoses() is run in CockpitLook. Get the last tracking pose obtained then.
+			// This pose was just used to render the current frame in xwingaliance.exe and will provide consistent tracking.			
+			vr::VRCompositor()->GetLastPoses(trackedDevicePoseArray, vr::k_unMaxTrackedDeviceCount, NULL, 0);
+			//log_debug("[DBG] ddraw.dll calling GetLastPoses()\n");
```

Comment:
Shouldn't there be similar changes in CockpitLook when g_bCorrectedHeadTracking is set/unset? I see you're reading the same setting ("pose_corrected_head_tracking") in cockpitlook.cpp, but other than setting a global variable, I'm not sure anything else is happening in CockpitLook.

Reply:
Nevermind, the relevant changes are in the HookCockpitLook project.